ui/pages/confirmations/send-legacy/send.utils.js

```diff
@@ -103,6 +106,27 @@ function generateERC20TransferData({
   );
 }
 
+function generateERC20ApprovalData({
+  spenderAddress = BURN_ADDRESS,
+  amount = '0x0',
+}) {
+  if (!spenderAddress) {
+    return undefined;
+  }
+  return (
+    TOKEN_APPROVAL_FUNCTION_SIGNATURE +
+    Array.prototype.map
+      .call(
+        encode(
+          ['address', 'uint256'],
+          [addHexPrefix(spenderAddress), addHexPrefix(amount)],
+        ),
+        (x) => `00${x.toString(16)}`.slice(-2),
+      )
+      .join('')
+  );
+}
+
```

Comment:
Avoid adding code to legacy files I think. This file has code to create data for send flows: ui/pages/confirmations/utils/send.ts